components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/defaultsequence/DefaultAuthSeqMgtServiceImpl.java

```diff
@@ -0,0 +1,613 @@
+                log.error(msg);
+            }
+            throw new DefaultAuthSeqMgtException(validationMsg.toArray(new String[0]));
+        }
+    }
+
+    private String removeUnsupportedXMLElements(String seqConfigXml) throws DefaultAuthSeqMgtException {
+
+        String updatedSeqConfigXml = null;
+        if (seqConfigXml != null) {
+            try {
+                DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+                InputSource src = new InputSource();
+                src.setCharacterStream(new StringReader(seqConfigXml));
+                Document doc = builder.parse(src);
+                NodeList nodeList = doc.getDocumentElement().getChildNodes();
+                for (int i = 0; i < nodeList.getLength(); i++) {
+                    Node currentNode = nodeList.item(i);
+                    if (currentNode.getNodeType() == Node.ELEMENT_NODE &&
+                            !currentNode.getNodeName().equals("AuthenticationSteps") &&
+                            !currentNode.getNodeName().equals("AuthenticationScript")) {
+                        doc.getDocumentElement().removeChild(currentNode);
+                    }
+                }
+
+                StringWriter stringWriter = new StringWriter();
+                Transformer transformer = TransformerFactory.newInstance().newTransformer();
+                transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "true");
+                transformer.transform(new DOMSource(doc), new StreamResult(stringWriter));
+                updatedSeqConfigXml = stringWriter.toString().replaceAll("(?m)^[ \t]*\r?\n", "");
+            } catch (ParserConfigurationException | SAXException | IOException | TransformerException e) {
+                throw new DefaultAuthSeqMgtServerException("Error when retrieving default authentication sequence", e);
+            }
+        }
+        return updatedSeqConfigXml;
+    }
+
+    private String marshalDefaultAuthSeq(LocalAndOutboundAuthenticationConfig sequence, String tenantDomain)
+            throws DefaultAuthSeqMgtException {
+
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(LocalAndOutboundAuthenticationConfig.class);
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            DocumentBuilderFactory docBuilderFactory = IdentityUtil.getSecuredDocumentBuilderFactory();
+            Document document = docBuilderFactory.newDocumentBuilder().newDocument();
+            marshaller.marshal(sequence, document);
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            transformer.setOutputProperty(OutputKeys.CDATA_SECTION_ELEMENTS,
```

Comment:
There are specific properties to be set in transformation factory as well. Verify with security team

Reply:
Added https://github.com/wso2/carbon-identity-framework/pull/1927/commits/65fad8a5f88765066dd229e55c21e8e605431c95#diff-02cca1637b3d4ec7d1d27ed55bf390eaR541